src/main/java/de/themoep/snap/forwarding/listener/ServerDisconnectListener.java

```diff
@@ -33,9 +33,6 @@ public ServerDisconnectListener(Snap snap) {
 
     @Subscribe(order = PostOrder.LAST)
     public void on(KickedFromServerEvent event) {
-        if (!event.getResult().isAllowed()) {
-            return;
-        }
```

Comment:
Tbh I'm not too sure about this event in particular or how exactly this behaves on Velocity when the kick is cancelled. I assume it simply keeps the player on the server?

If so then it might be good to have that check and not call the disconnect I guess? Maybe something to test/think about but the other stuff looks good to me now, thanks!